### PR TITLE
8260606: Update Valhalla core-libs naming for methods related to primitive classes

### PR DESCRIPTION
**Pull request by RogerRiggs**
Update terminology used to refer to primitive classes (from inline classes).
Drop the descriptions using "projection" and use "conversion" as needed.
Updated dependencies on Class.isPrimitiveClass().

[Updating test names will be a separate task.]

---------
### Progress
- [x] Change must not contain extraneous whitespace

### Issue
 * [JDK-8260606](https://bugs.openjdk.java.net/browse/JDK-8260606): Update Valhalla core-libs naming for methods related to primitive classes

### Reviewers
 * [Mandy Chung](https://openjdk.java.net/census#mchung) (@mlchung - Committer)

### Download
`$ git fetch https://git.openjdk.java.net/valhalla pull/318/head:pull/318`
`$ git checkout pull/318`

### CODE REVIEW

File: src/java.base/share/classes/java/lang/Class.java [hunk 1]
```diff
@@ -554,137 +553,129 @@ static String typeVarBounds(TypeVariable<?> typeVar) {
     }
 
     /**
-     * Returns {@code true} if this class is an inline class.
+     * Returns {@code true} if this class is a primitive class.
      *
-     * @return {@code true} if this class is an inline class
+     * @return {@code true} if this class is a primitive class, otherwise false
```
Comment (mlchung) [inline src/java.base/share/classes/java/lang/Class.java:558]: Nit: s/false/`{@code false}`/

File: src/java.base/share/classes/java/lang/Class.java [hunk 2]
```diff
@@ -554,137 +553,129 @@ static String typeVarBounds(TypeVariable<?> typeVar) {
...
     /**
-     * Returns a {@code Class} object representing the <em>value projection</em>
-     * type of this class if this {@code Class} represents the reference projection
-     * type of an {@linkplain #isInlineClass() inline class}.
-     * If this {@code Class} represents the value projection type
-     * of an inline class, then this method returns this class.
+     * Returns a {@code Optional<Class>} object representing the <em>primitive value type</em>
+     * of this class if this {@code Class} represents the <em>primitive reference type</em>
+     * of a {@linkplain #isPrimitiveClass() primitive class}.
+     * If this {@code Class} represents the primitive value type
```
Comment (mlchung) [inline src/java.base/share/classes/java/lang/Class.java:569]: "the primitive reference type of a primitive class" perhaps should be shortened to 
"the reference type of a primitive class"
"the value type of a primitive class" similarly for value type.

File: src/java.base/share/classes/java/lang/Class.java [hunk 3]
```diff
@@ -554,137 +553,129 @@ static String typeVarBounds(TypeVariable<?> typeVar) {
...
+     * primitive reference type, if present.
      *
-     * If this Class object is neither a value projection type nor
-     * a reference projection type for an inline class, then an empty array
+     * If this Class object is neither a primitive value type nor
+     * a primitive reference type for a primitive class, then an empty array
      * is returned.
      */
-    private Class<?>[] newProjectionTypeArray() {
+    private Class<?>[] newPrimitiveTypeArray() {
```
Comment (mlchung) [inline src/java.base/share/classes/java/lang/Class.java:666]: perhaps `newTypeArrayForPrimitiveClass`

File: src/java.base/share/classes/java/lang/invoke/ValueBootstrapMethods.java [hunk 4]
```diff
@@ -552,17 +552,17 @@ private static LinkageError newLinkageError(Throwable e) {
     };
 
     /**
-     * Invoke the bootstrap methods hashCode for the given inline object.
+     * Invoke the bootstrap methods hashCode for the given primitive classobject.
```
Comment (mlchung) [inline src/java.base/share/classes/java/lang/invoke/ValueBootstrapMethods.java:555]: a space is missing in `classobject`

File: src/java.base/share/classes/java/lang/invoke/ValueBootstrapMethods.java [hunk 5]
```diff
@@ -552,17 +552,17 @@ private static LinkageError newLinkageError(Throwable e) {
...
      * @param o the instance to hash.
-     * @return the hash code of the given inline object.
+     * @return the hash code of the given primitive classobject.
```
Comment (mlchung) [inline src/java.base/share/classes/java/lang/invoke/ValueBootstrapMethods.java:557]: space is missing in the word `classobject`

File: src/java.base/share/classes/java/lang/invoke/ValueBootstrapMethods.java [hunk 6]
```diff
@@ -579,17 +579,17 @@ private static int inlineObjectHashCode(Object o) {
     };
 
     /**
-     * Invoke the bootstrap methods hashCode for the given inline object.
+     * Invoke the bootstrap methods hashCode for the given primitive classobject.
      * @param o the instance to hash.
-     * @return the string representation of the given inline object.
+     * @return the string representation of the given primitive classobject.
```
Comment (mlchung) [inline src/java.base/share/classes/java/lang/invoke/ValueBootstrapMethods.java:584]: same typo `classobject`

File: src/java.base/share/classes/jdk/internal/misc/Unsafe.java [hunk 7]
```diff
@@ -247,46 +247,46 @@ public boolean isFlattened(Field f) {
...
+        if (ref == null && pc.isPrimitiveClass()) {
+            // If the type of the returned reference is a regular primitive type
             // return an uninitialized default value if null
-            ref = uninitializedDefaultValue(vc);
+            ref = uninitializedDefaultValue(pc);
         }
         return ref;
     }
 
     public Object getReferenceVolatile(Object o, long offset, Class<?> vc) {
```
Comment (mlchung) [inline src/java.base/share/classes/jdk/internal/misc/Unsafe.java:276]: `vc` parameter should also be renamed to `pc` to be consistent with other methods.

**Review by mlchung — APPROVED** [review]
Looks good.   Do you want to add `IdentityObject` in a separate PR?  It's okay with me to add this in this PR.

**Comment by RogerRiggs** [comment]
> Looks good. Do you want to add `IdentityObject` in a separate PR? It's okay with me to add this in this PR.

IdentityObject is already in java.lang.

java.lang.PrimitiveObject has not yet been added.

**Comment by mlchung** [comment]
> > Looks good. Do you want to add `IdentityObject` in a separate PR? It's okay with me to add this in this PR.
> 
> IdentityObject is already in java.lang.
> 
> java.lang.PrimitiveObject has not yet been added.

I meant `PrimitiveObject` (sorry for the typo).

**Review by mlchung — APPROVED** [review]

**Pull request closed without merging**